Controller/Redirect/In.php

```diff
@@ -86,86 +133,81 @@ public function __construct(
 
-        /** Fix for card payments, as they don't have order in session */
-        if (empty($order->getData())) {
-            $quote = $this->checkoutSession->getQuote();
-            if (!empty($quote->getEntityId())) {
-                $orders = $this->orderService->getAllOrdersByQuote($quote);
-                $order = end($orders);
-                $this->checkoutSession->setData('last_real_order_id', $order->getIncrementId());
-            }
-        }
+        $rvvupPaymentId = $payment->getAdditionalInformation(Method::PAYMENT_ID);
 
-        if (!$order->getEntityId() || $order->getPayment() === null) {
+        try {
+            $orderId = $this->quoteManagement->placeOrder($quote->getEntityId(), $payment);
+        } catch (\Exception $e) {
             $this->logger->error(
-                'Could not find ' . (!$order->getEntityId() ? 'order' : 'payment') . ' for the checkout session',
+                'Order placement within rvvup payment failed',
                 [
-                    'order_id' => $order->getEntityId()
+                    'payment_id' => $payment->getEntityId(),
+                    'last_transaction_id' => $lastTransactionId,
+                    'rvvup_order_id' => $rvvupId,
+                    'message' => $e->getMessage()
                 ]
             );
             $this->messageManager->addErrorMessage(
                 __(
-                    'An error occurred while processing your payment (ID %1). Please contact us.',
+                    'An error occurred while creating your order (ID %1). Please contact us.',
                     $rvvupId
                 )
             );
-
             return $this->redirectToCart();
         }
-
-        // Now validate that last payment transaction ID matches the Rvvup ID.
-        $lastTransactionId = $order->getPayment()->getLastTransId();
-
-        if ($rvvupId !== $lastTransactionId) {
+        try {
+            $this->sdkProxy->paymentCapture($lastTransactionId, $rvvupPaymentId);
+        } catch (\Exception $e) {
             $this->logger->error(
-                'Payment ID when redirecting from Rvvup checkout does not match order in checkout session',
+                'Order placement failed during payment capture',
                 [
-                    'order_id' => $order->getEntityId(),
-                    'payment_id' => $order->getPayment()->getEntityId(),
+                    'payment_id' => $payment->getEntityId(),
                     'last_transaction_id' => $lastTransactionId,
-                    'rvvup_order_id' => $rvvupId
+                    'rvvup_order_id' => $rvvupId,
+                    'message' => $e->getMessage()
                 ]
             );
             $this->messageManager->addErrorMessage(
                 __(
-                    'An error occurred while processing your payment (ID %1). Please contact us.',
+                    'An error occurred while capturing your order (ID %1). Please contact us.',
```

Comment:
`capturing your payment`

I think we should keep the "payment" naming over order when we refer to rvvup objects as we are moving away from order terminology generally.